packages/client/src/features/forum/components/NewMessageModal.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  createThreadMessages,
+  getThreadMessages,
+} from '@/app/forum/forumSlice';
+import { AppDispatch } from '@/app/store';
+import DataField, { DATA_FIELD_VARIANTS } from '@/components/DataField';
+import MainButton from '@/components/MainButton';
+import ModalWindow, { ModalProps } from '@/components/ModalWindow';
+import { Stack } from '@mui/material';
+import { FunctionComponent, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useParams, useSearchParams } from 'react-router-dom';
+
+export const NewMessageModal: FunctionComponent<ModalProps> = ({
+  open,
+  handleClose,
+  handleOpen,
+  title,
+}) => {
+  const dispatch = useDispatch<AppDispatch>();
+  const [, setSearchParams] = useSearchParams();
+  const [message, setMessage] = useState<string>('');
+  const { id = '' } = useParams();
+
+  const onCreate = async () => {
+    if (!message) {
+      return;
+    }
+    try {
+      //@ts-ignore
+      await dispatch(createThreadMessages({ TopicId: id, content: message }));
+
+      handleClose();
+
+      await dispatch(getThreadMessages(id));
+
+      setSearchParams({ page: '1' });
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  return (
+    <ModalWindow {...{ open, handleClose, handleOpen, title }}>
```

Comment:
Это такое сокращение для 
```suggestion
    <ModalWindow open={open} handleClose={handleClose} handleOpen={handleOpen} title={title}>
```
?

Если да, то оно не стоит того, будущие возможные проблемы с дебагом подобных мест перекроют все плюсы